### PR TITLE
add base api

### PR DESCRIPTION
**Pull request by AnatoliAgeichik**
In this PR was added 
- model(Singer and Track)
- base api(view all tracks, singers; every tracks, singers; and  added possibility  translate tracks
- add permission

### CODE REVIEW

File: song_translator/track/apps.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
+
+
+class SongConfig(AppConfig):
+    default_auto_field = 'django.db.models.BigAutoField'
```
Comment (artsiom-streetshares) [inline song_translator/track/apps.py:5]: why you need this?

File: song_translator/track/views.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class TrackTranslate(generics.RetrieveUpdateDestroyAPIView):
+
+    queryset = Track.objects.all()
+    serializer_class = TrackSerializer
+
+    def get(self, request, pk, lang):
+        translator = google_translator()
+        return Response(translator.translate(self.get_object().text, lang_tgt=lang))
```
Comment (artsiom-streetshares) [inline song_translator/track/views.py]: hmm, that means we are translating song every time on issuing GET request, doesn't sound good to me. Let's rather translate at the moment of Track creation and save the translation to the database.

**Comment by artsiom-streetshares** [comment]
Let's have translation as a model as well. It will have foreign key to track, translation_text and target language. And then we will present users with an ability to add translations. They will be able to enter full translation by themselves, or use automatic translation or use automatic translation and edit it slightly manually.

File: song_translator/track/urls.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from django.urls import path
+from rest_framework.urlpatterns import format_suffix_patterns
+
+from .views import TrackList, TrackDetail, SingerList, SingerDetail, TrackTranslate
+
+
+urlpatterns = [
+    path('tracks/', TrackList.as_view()),
+    path('tracks/<int:pk>/', TrackDetail.as_view()),
+    path('tracks/<int:pk>/<str:lang>', TrackTranslate.as_view()),
```
Comment (artsiom-streetshares) [inline song_translator/track/urls.py]: let's restrict languages that could be used in urls for example like this. https://stackoverflow.com/a/6253248/3888551

File: song_translator/track/models.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    def __str__(self):
+        return self.name
+
+
+class Track(models.Model):
+    name = models.CharField(max_length=64)
+    text = models.TextField()
+    translate_text = models.TextField(blank=True)
+    original_language = models.CharField(max_length=2)
```
Comment (artsiom-streetshares) [inline song_translator/track/models.py]: let's try restricting languages we support via enum https://www.postgresql.org/docs/9.1/datatype-enum.html
maybe also on serializer level

File: song_translator/song_translator/settings.py [hunk 5]
```diff
@@ -78,7 +99,7 @@
 DATABASES = {
  'default': {
         'ENGINE': 'django.db.backends.postgresql_psycopg2',
-        'NAME': 'song_translator',
+        'NAME': 'translatorr',
```
Comment (artsiom-streetshares) [inline song_translator/song_translator/settings.py]: why double r at the end, is that a typo?

File: song_translator/track/models.py [hunk 6]
```diff
@@ -11,10 +52,18 @@ def __str__(self):
...
+    original_language = models.CharField(max_length=2, choices=LANG_CHOICES, default="en")
     singer = models.ManyToManyField(Singer)
-    owner = models.ForeignKey('auth.User', related_name='tracks', on_delete=models.CASCADE, default=1)
+    owner = models.ForeignKey('track.TranslatorUser', related_name='tracks', on_delete=models.CASCADE, default=1)
 
     def __str__(self):
         return self.name
+
+
+class Translate(models.Model):
```
Comment (artsiom-streetshares) [inline song_translator/track/models.py]: Use nouns for table, i.e. Translation

File: song_translator/track/urls.py [hunk 7]
```diff
@@ -1,13 +1,19 @@
-from django.urls import path
+from django.urls import path, re_path
 from rest_framework.urlpatterns import format_suffix_patterns
 
-from .views import TrackList, TrackDetail, SingerList, SingerDetail, TrackTranslate
-
+from .views import TrackList, TrackDetail, SingerList, SingerDetail, track_translate
 
+lang = ['af', 'sq', 'am', 'ar', 'hy', 'az', 'eu', 'be', 'bn', 'bs', 'bg', 'ca', 'ceb', 'ny', 'co',
```
Comment (artsiom-streetshares) [inline song_translator/track/urls.py]: could you define your languages in one place? currently it's also defined in utils

File: song_translator/track/views.py [hunk 8]
```diff
@@ -19,14 +22,40 @@ class TrackDetail(generics.RetrieveUpdateDestroyAPIView):
     permission_classes = [permissions.IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly]
 
 
-class TrackTranslate(generics.RetrieveUpdateDestroyAPIView):
+@api_view(['GET', 'PUT', 'DELETE'])
```
Comment (artsiom-streetshares) [inline song_translator/track/views.py]: how do you create translations?

Comment (AnatoliAgeichik) [reply song_translator/track/views.py]: when we do not have the  translation, it is automatically created. That is, if nothing is found in the get request, an automatic translation is created

File: song_translator/track/utils.py [hunk 9]
```diff
@@ -105,3 +105,11 @@
     ('yo', 'yoruba'),
     ('zu', 'zulu'),
 )
+lang_short = ['af', 'sq', 'am', 'ar', 'hy', 'az', 'eu', 'be', 'bn', 'bs', 'bg', 'ca', 'ceb', 'ny', 'co',
```
Comment (artsiom-streetshares) [inline song_translator/track/utils.py:108]: still duplication. could you extract list of tuples from Choices constructor to a variable and then use it for both `lang_re` and constructor of Choices?

File: song_translator/track/views.py [hunk 10]
```diff
@@ -22,22 +22,41 @@ class TrackDetail(generics.RetrieveUpdateDestroyAPIView):
...
+                                                                         'original_language'])
+        serializer = TranslateSerializer(data=translation_data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    elif request.method == 'DELETE':
+        count = Translation.objects.filter(track_id=pk).delete()
+        return Response({'message': '{} translations were deleted successfully!'.format(count[0])},
```
Comment (artsiom-streetshares) [inline song_translator/track/views.py:51]: use f-strings instead

File: song_translator/track/migrations/0005_delete_comment.py [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('track', '0004_rename_translate_translation'),
+    ]
+
+    operations = [
+        migrations.DeleteModel(
```
Comment (artsiom-streetshares) [inline song_translator/track/migrations/0005_delete_comment.py:13]: that's fine for now, but in real world strive for least possible number of migrations per PR. I mean you create and delete Comment model in one PR

**Review by artsiom-streetshares — APPROVED** [review]
LG. Let's fix comments in the next PR

**Pull request merged**